keyboards/aplusx_pulsar/xboard_qs/xboard_qs.c

```diff
@@ -0,0 +1,256 @@
+/* Copyright 2024 Aplusx Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "xboard_qs.h"
+
+static bool     INIT3S_on = false;
+static bool     INIT3S_led_on = false;
+static uint16_t INIT3S_timer;           // for custom key, INIT_3S
+
+static bool     kvm_pc_sel      = false;
+static uint16_t kvm_timer;
+static bool     kvm_sel_on      = false;
+
+static bool isMacMode = false;
+static uint8_t Status_LED_Bright;
+#define kvm_deadtime    100
+
+typedef union {
+    uint32_t raw;
+    struct {
+        bool    eeprom_kvm_pc_sel :1;
+        uint8_t eeprom_Status_LED_Bright;       //
+    };
+} keyboard_config_t;
+
+keyboard_config_t keyboard_config;
+
+void kvm_switch(bool pc_num){
+    gpio_write_pin_high(GPIO_KM_OE);
+    gpio_write_pin_low(GPIO_KM_PWEN);
+    if (!pc_num) gpio_write_pin_low(GPIO_KM_SEL);
+    else   gpio_write_pin_high(GPIO_KM_SEL);
+    gpio_write_pin_low(GPIO_KM_OE);
+
+    kvm_timer = timer_read();
+    kvm_sel_on = true;
+};
+
+void keyboard_pre_init_kb(void) {
+    gpio_set_pin_output(GPIO_KM_OE);       // H/W power on default = 0
+    gpio_set_pin_output(GPIO_KM_SEL);      // H/W power on default = 0
+    gpio_set_pin_output(GPIO_KM_PWEN);     // H/W power on default = 1
+}
+
+void keyboard_post_init_kb(void) {
+    keyboard_config.raw = eeconfig_read_kb();                 // Read the config from EEPROM
+    kvm_pc_sel      = keyboard_config.eeprom_kvm_pc_sel;
+    kvm_switch(kvm_pc_sel);
+
+    Status_LED_Bright = keyboard_config.eeprom_Status_LED_Bright;   // 0 = 초기값, 1~5 = 50~250의 5단계 밝기
+    if (Status_LED_Bright == 0) Status_LED_Bright = 5;
+}
+
+#define HCS(report) host_consumer_send(record->event.pressed ? report : 0); return false
+#define HSS(report) host_system_send(record->event.pressed ? report : 0); return false
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) { return false; }
+    switch (keycode) {
+        case INIT_3S:
+            if (record->event.pressed){
+                INIT3S_timer = timer_read();
+                INIT3S_on    = true;
+            }
+            else{
+                INIT3S_on   = false;
+            }
+            break;
+
+        case KC_SPOTLIGHT:
+            HCS(0x221);
```

Comment:
`KC_WWW_SEARCH` already does this (uses this consumer code).

Reply:
I got it. thank you